WDE.SmartScriptEditor/Editor/ViewModels/SmartDataEditors/SmartDataParameterEditorViewModel.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Prism.Mvvm;
+using Prism.Commands;
+using WDE.Common.Parameters;
+using WDE.SmartScriptEditor.Data;
+using WDE.Common.ViewHelpers;
+using System.Collections.ObjectModel;
+using WDE.Common.Managers;
+
+namespace WDE.SmartScriptEditor.Editor.ViewModels
+{
+    class SmartDataParameterEditorViewModel: IDialog
+    {
+        // TEMP until better solution comes to my mind ~Pelekon
+        private static readonly Dictionary<string, bool> NotStoredParameters = new Dictionary<string, bool>
+        {
+            { "Parameter", false },
+            { "PercentageParameter", false },
+            { "SwitchParameter", true },
+            { "FlagParameter", true },
+        };
+        public SmartDataParameterEditorViewModel(IParameterFactory parameterFactory, in SmartParameterJsonData source)
+        {
+            // Params loaded from DBC etc + basic parameters defined my module itself e.g.CreatureParameter
+            var primitiveParameters = parameterFactory.GetKeys().Where(x => !NotStoredParameters.ContainsKey(x)).ToList();
+            ParameterKeys = new List<SmartDataParamterData>(NotStoredParameters.Count + primitiveParameters.Count);
+            foreach (var param in NotStoredParameters)
+                ParameterKeys.Add(new SmartDataParamterData(param.Key, param.Value));
+            foreach (var param in primitiveParameters)
+                ParameterKeys.Add(new SmartDataParamterData(param, false));
+            ParameterKeys.Sort();
+            Source = new SmartDataParameterEditorData(in source);
+            AreValuesVisible = false;
+            AssignCurrentParamKey();
+            AddValue = new DelegateCommand(AddValueToParam);
+            DeleteValue = new DelegateCommand(DeleteValueFromParam);
+            SaveValue = new DelegateCommand(() => CloseOk?.Invoke());
+        }
+
+        public SmartDataParameterEditorData Source { get; }
+        public bool AreValuesVisible { get; private set; }
```

Comment:
Now, if this value change (for instance when changing parameter type), the UI doesn’t update. If you implement INotifyPropertyChanged for this type, will it update?